crates/bls-gadgets/src/hash_to_group.rs

```diff
@@ -75,11 +75,14 @@ pub struct HashToGroupGadget<P> {
 // by taking the input, compressing it via an instantiation of Pedersen Hash with a CRH over Edwards SW6
 // and then hashing it to bits and to group
 impl HashToGroupGadget<Bls12_377_Parameters> {
+    /// Returns the G1 constrained hash of the message with the provided counter along with auxiliary
+    /// data which can be used by consumers to split the calculation in 2 SNARKs as detailed in ... TODO
+    #[allow(clippy::type_complexity)]
     pub fn enforce_hash_to_group<CS: ConstraintSystem<Bls12_377_Fq>>(
```

Comment:
enforce_hash_to_group shouldn't call the gadget version of hash_to_bits. This will make it generate constraints for the blake2 hash in SW6, where we want these constraints only in the BLS12-377 proofs. Instead, this function should use the output from the pedersen hash, and natively compute the expected xof bits.